thrust/device_vector.h

```diff
@@ -69,6 +69,12 @@ template<typename T, typename Alloc = thrust::device_malloc_allocator<T> >
     device_vector(void)
       :Parent() {}
 
+    /*! Define an empty destructor to explicitly specify
```

Comment:
It's fine to leave a note to say why we're defining these destructors (i.e. to work around an nvcc warning), but I'm not sure it makes sense for it to show up in the Doxygen-generated documentation. (The `/*!` syntax causes the comment to be visible via Doxygen.)

If a user read this in the documents on the web, the stuff about execution space would probably just cause confusion. So, I think the user-visible documentation should just say something similar to what is said here: http://en.cppreference.com/w/cpp/container/vector/~vector.